src/rgw/rgw_restore.cc

```diff
@@ -749,4 +750,151 @@ int Restore::restore_obj_from_cloud(rgw::sal::Bucket* pbucket,
   return ret;
 }
 
+int Restore::list(const DoutPrefixProvider* dpp, rgw::sal::Driver* driver,
+                  RestoreEntry& entry, std::optional<string> restore_status_filter,
+                  std::string& err_msg, RGWFormatterFlusher& flusher, optional_yield y)
+{
+  int ret = 0;
+  std::unique_ptr<rgw::sal::Bucket> bucket;
+  ret = driver->load_bucket(dpp, entry.bucket, &bucket, null_yield);
```

Comment:
shouldn't this be called with `y` yield instead of hard-coding `null_yield` here. Same applies for all other calls.